hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -61,6 +61,28 @@ def length(self):
         return self.end - self.start
 
 
+@attr.s(slots=True)
+class Block(object):
```

Comment:
We haven't done this for `Example` so it's up to you if you want to do it here, but it occurs to me that it would be good to have docstrings explaining what these things represent.

Reply:
I won't do this right now, but it's a good suggestion for later.